src/control/createIndex.ts

```diff
@@ -98,52 +85,134 @@ export interface CreateIndexPodSpec {
   sourceCollection?: string;
 }
 
-const CreateIndexOptionsSchema = Type.Object(
-  {
-    name: IndexNameSchema,
-    dimension: DimensionSchema,
-    metric: MetricSchema,
-    deletionProtection: Type.Optional(Type.String()),
-    spec: Type.Object({
-      serverless: Type.Optional(
-        Type.Object({
-          cloud: CloudSchema,
-          region: RegionSchema,
-        })
-      ),
-
-      pod: Type.Optional(
-        Type.Object({
-          environment: EnvironmentSchema,
-          replicas: Type.Optional(ReplicasSchema),
-          shards: Type.Optional(ShardsSchema),
-          podType: Type.Optional(PodTypeSchema),
-          pods: Type.Optional(PodsSchema),
-          metadataConfig: Type.Optional(MetadataConfigSchema),
-          sourceCollection: Type.Optional(CollectionNameSchema),
-        })
-      ),
-    }),
-
-    waitUntilReady: Type.Optional(Type.Boolean()),
-    suppressConflicts: Type.Optional(Type.Boolean()),
-  },
-  { additionalProperties: false }
-);
-
 export const createIndex = (api: ManageIndexesApi) => {
-  const validator = buildConfigValidator(
-    CreateIndexOptionsSchema,
-    'createIndex'
-  );
+  const validator = async (options: CreateIndexOptions) => {
+    if (options) {
+      ValidateProperties(options, [
```

Comment:
While I appreciate the thoroughness in evaluating the keys that were possibly passed, I think maintaining these inline representations of the types keys is going to be prone to problems when the types shift around. Can we more tightly couple the lists that are being passed to validate things with the type? I'd look at using `keysOf` and generics if possible: https://www.executeprogram.com/courses/advanced-typescript/lessons/keyof-generic-constraints